### PR TITLE
YLP-1531 medical files widget

### PR DESCRIPTION
**Pull request by fatb38**

### CODE REVIEW

File: packages/yourloops/components/dialogs/medical-record-delete-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    >
+      <DialogTitle>
+        <Typography variant="h5">
+          {t("delete-medical-report")}
+        </Typography>
+      </DialogTitle>
+
+      <DialogContent>
+        <DialogContentText>
+          {t("delete-warning")} {t("medical-record-pdf")}{new Date(medicalRecord.creationDate).toLocaleDateString()} ?
```
Comment (TimCanu) [inline packages/yourloops/components/dialogs/medical-record-delete-dialog.tsx:90]: This translation key (delete-warning) should handle parameters in order to properly translate the "?" (there is a space before it in French, but not in English

File: packages/yourloops/components/dashboard-widgets/medical-files/medical-files-widget.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  const team = getRemoteMonitoringTeams().find(team => team.members.find(member => member.user.userid === patient.userid));
+
+  if (!team) {
+    throw Error(`Could not find monitoring team for patient with id: ${patient.userid}`);
+  }
+
+  return (
+    <React.Fragment>
+      <Card className={classes.medicalFilesWidget} id="medical-files-card">
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/medical-files/medical-files-widget.tsx:81]: opt: rename the component to "medical-files-card"

File: packages/yourloops/components/dashboard-widgets/medical-files/medical-files-widget.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  const { patient } = props;
+
+  const team = getRemoteMonitoringTeams().find(team => team.members.find(member => member.user.userid === patient.userid));
+
+  if (!team) {
+    throw Error(`Could not find monitoring team for patient with id: ${patient.userid}`);
+  }
+
+  return (
+    <React.Fragment>
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/medical-files/medical-files-widget.tsx:80]: Can you explain why it is useful to have a <React.Fragment>?

File: packages/yourloops/components/dashboard-widgets/medical-files/medical-record-list.tsx [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+ */
+
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { makeStyles, Theme } from "@material-ui/core/styles";
+
+import CreateOutlinedIcon from "@material-ui/icons/CreateOutlined";
+import DescriptionOutlinedIcon from "@material-ui/icons/DescriptionOutlined";
+import NoteAddIcon from "@material-ui/icons/NoteAdd";
+import TrashCanOutlined from "../../icons/TrashCanOutlined";
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/medical-files/medical-record-list.tsx:35]: Not wanting to be annoying but this import is not at the right place

File: packages/yourloops/components/dashboard-widgets/medical-files/prescription-list.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { makeStyles, Theme } from "@material-ui/core/styles";
+
+import FileChartOutlinedIcon from "../../icons/FileChartOutlinedIcon";
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/medical-files/prescription-list.tsx:32]: again, not in right spot

File: packages/yourloops/components/dashboard-widgets/medical-files/weekly-report-list.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { makeStyles, Theme } from "@material-ui/core/styles";
+
+import FileChartOutlinedIcon from "../../icons/FileChartOutlinedIcon";
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/medical-files/weekly-report-list.tsx:32]: :wink:

File: packages/yourloops/components/dashboard-widgets/medical-files/weekly-report-list.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+export default function WeeklyReportList({ teamId, patientId }: CategoryProps): JSX.Element {
+  const { t } = useTranslation("yourloops");
+  const classes = useStyle();
+  const [weeklyReports, setWeeklyReports] = useState<WeeklyReport[]>([]);
+  const [hoveredItem, setHoveredItem] = useState<string | undefined>(undefined);
+
+  useEffect(() => {
+    (async () => {
+      setWeeklyReports(await MedicalFilesApi.getWeeklyReports(patientId, teamId));
+      console.log(await MedicalFilesApi.getWeeklyReport(patientId, teamId, "a35a73e7-3666-4b80-9cac-d7c4d83a4a4f"));
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/medical-files/weekly-report-list.tsx:74]: to be removed

File: packages/yourloops/components/dialogs/medical-record-delete-dialog.tsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { makeStyles } from "@material-ui/core/styles";
```
Comment (TimCanu) [inline packages/yourloops/components/dialogs/medical-record-delete-dialog.tsx:30]: imports not conform to our "import strategy"

File: packages/yourloops/components/dialogs/medical-record-edit-dialog.tsx [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+import Box from "@material-ui/core/Box";
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import Divider from "@material-ui/core/Divider";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import TextField from "@material-ui/core/TextField";
+import Typography from "@material-ui/core/Typography";
```
Comment (TimCanu) [inline packages/yourloops/components/dialogs/medical-record-edit-dialog.tsx:42]: "import strategy"

File: packages/yourloops/components/dialogs/medical-record-edit-dialog.tsx [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+          onClick={onClose}
+        >
+          {t("cancel")}
+        </Button>
+        <ProgressIconButtonWrapper inProgress={inProgress}>
+          <Button
+            variant="contained"
+            color="primary"
+            disableElevation
+            disabled={readonly || inProgress || !diagnosis && !trainingSubject && !progressionProposal}
```
Comment (TimCanu) [inline packages/yourloops/components/dialogs/medical-record-edit-dialog.tsx:200]: to handle with useMemo?

File: packages/yourloops/components/icons/FileChartOutlinedIcon.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2022, Diabeloop
+ * Rounded Hospital Icon
```
Comment (TimCanu) [inline packages/yourloops/components/icons/FileChartOutlinedIcon.tsx:3]: wrong

File: packages/yourloops/components/icons/TrashCanOutlined.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2022, Diabeloop
+ * Rounded Hospital Icon
```
Comment (TimCanu) [inline packages/yourloops/components/icons/TrashCanOutlined.tsx:3]: liar

File: packages/yourloops/components/dashboard-widgets/medical-files/medical-files-widget.tsx (hunk 3 above)
Comment (fatb38) [reply packages/yourloops/components/dashboard-widgets/medical-files/medical-files-widget.tsx:80]: just forgot to remove it !

File: packages/yourloops/components/dashboard-widgets/medical-files/medical-files-widget.tsx (hunk 2 above)
Comment (fatb38) [reply packages/yourloops/components/dashboard-widgets/medical-files/medical-files-widget.tsx:81]: I named it just like others widget (ex: patient-info-widget). What do you think ?

Comment (TimCanu) [reply packages/yourloops/components/dashboard-widgets/medical-files/medical-files-widget.tsx]: I think that the main component is a Card so the component should be medical-files-card (as for alarm-card). But this is my opinion :)

File: packages/yourloops/components/dialogs/medical-record-delete-dialog.tsx (hunk 8 above)
Comment (TimCanu) [reply packages/yourloops/components/dialogs/medical-record-delete-dialog.tsx:30]: I thought that all material stuff was in the middle section but maybe I don't know everything :slightly_frowning_face: 

But anyway, the ere should be a line break after the Typography import

**Review by TimCanu — APPROVED** [review]

**Comment by TimCanu** [comment]
Done in https://github.com/mdblp/blip/pull/841

**Pull request closed without merging**